restapi.go

```diff
@@ -1311,7 +1311,49 @@ func (s *Session) ChannelMessageSendComplex(channelID string, data *MessageSend)
 		data.Embed.Type = "rich"
 	}
 
-	response, err := s.RequestWithBucketID("POST", EndpointChannelMessages(channelID), data, EndpointChannelMessages(channelID))
+	var response []byte
+	if data.File != nil {
+		body := &bytes.Buffer{}
+		bodywriter := multipart.NewWriter(body)
+
+		// What's a better way of doing this? Reflect? Generator? I'm open to suggestions
+
+		if err = bodywriter.WriteField("content", data.Content); err != nil {
+			return nil, err
+		}
+		var embed []byte
+		embed, err = json.Marshal(data.Embed)
+		if err != nil {
```

Comment:
You're only writing the embed on failure, this embed write shouldn't be in an if scope.